community-tutorials/web/01-en.md

```diff
@@ -0,0 +1,404 @@
+author_bio: -
+tags: [web, mysql, linux, debian, mariadb, phpmyadmin, nextcloud, wordpress, yourls]
+netcup_product_url: https://netcup.de/bestellen/produkt.php?produkt=2623
+language: en
+available_languages: en
+---
+
+# Introduction
+This tutorial will show you, how you can install a Web Server (Apache) + PHP + MariaDB (MySQL) + Certbot + PHPMyAdmin + Wordpress + YOURLS + Nextcloud + Uptime Kuma on Debian <br>
+I am doing this on a Debian 11 server (RS 2000), but it also should work on other Debian versions<br>
+This tutorial contains content of https://github.com/SanCraftDev/Debian-Setup/blob/main/README.md and https://github.com/louislam/uptime-kuma
+
+# Requirements
+You need a Debian Server with an internet connection, which is accessible from the internet over a static IP on port 80 and 443.<br>
+You also need root permissions on your server.
+
+# Step 1 - Apache2
+
+## Step 1.1 - Installing Snapd and Certbot
+First we need to install Snapd and Certbot to get SSL Certificates
+```sh
+apt update && apt upgrade -y && apt autoremove -y
+apt install curl sudo wget apache2-utils nano git snapd cron zip unzip tar screen lsb_release -y
+snap install core
+snap install core; sudo snap refresh core
+sudo snap install --classic certbot
+sudo ln -s /snap/bin/certbot /usr/bin/certbot
+{ crontab -l 2>/dev/null; echo "$(( $RANDOM % 60 )) $(( $RANDOM % 3 + 3 )) * * * sudo certbot renew --dry-run" ; } | crontab -
+```
+
+## Step 1.2 - Installing Apache2
+Now we are going to install apache2
+```sh
+apt install apache2 libapache2-mod-php -y
+```
+Now we are going to configure apache2
+```sh
+a2enmod rewrite
+a2enmod headers
+a2enmod env
+a2enmod dir
+a2enmod mime
+a2enmod proxy
+a2enmod proxy_http
+a2enmod proxy_wstunnel
+a2enmod headers
+a2enmod ssl
+```
+```sh
+nano /etc/apache2/apache2.conf
+```
+Now search with ctrl + w `<Directory /var/www/>` <br>
+Now go some lines down to `AllowOverride None` and replace it with `AllowOverride All` <br>
+Save your edit with ctrl + x, then press y and now save it with enter and restart apache
+```sh
+service apache2 restart
+```
+
+## Step 1.3 Installing PHP
+```sh
+wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg
+sh -c 'echo "deb https://packages.sury.org/php/ $(lsb_release -sc) main" > /etc/apt/sources.list.d/php.list'
+apt update
+apt install zip unzip redis-server libapache2-mod-php8.0 libmagickcore-6.q16-6-extra php-dompdf php-pear php8.0 php8.0-{cli,gd,mysql,pdo,mbstring,tokenizer,bcmath,xml,fpm,curl,zip,common,intl,redis,opcache,readline,xsl,bz2,imagick,bcmath,gmp,sqlite3,apcu} -y
+nano /etc/php/8.0/apache2/php.ini
+```
+Add now in the first free line: `apc.enable_cli = 1` <br>
+Now press ctrl + w and enter: `memory_limit = 128M` now replace this line with `memory_limit = 1024M` or use more/less ram <br>
+Now press ctrl + w and enter: `post_max_size = 8M` now replace this line with `post_max_size = 10G` - defines the maximum upload file size <br>
```

Comment:
G stands for gigabyte, and it does not need to be so high, but it can be possible that someone wants to upload a 10 gigabyte file to his nextcloud for examle